collector/info_schema_query_response_time.go

```diff
@@ -12,42 +12,55 @@ import (
 	"github.com/prometheus/common/log"
 )
 
-const (
-	queryResponseCheckQuery = `SELECT @@query_response_time_stats`
-	queryResponseTimeQuery  = `
-		SELECT
-		    TIME, COUNT, TOTAL
-		  FROM information_schema.query_response_time
-		`
-)
+const queryResponseCheckQuery = `SELECT @@query_response_time_stats`
 
 var (
-	infoSchemaQueryResponseTimeCountDesc = prometheus.NewDesc(
-		prometheus.BuildFQName(namespace, informationSchema, "query_response_time_count"),
-		"The number of queries according to the length of time they took to execute.",
-		[]string{}, nil,
-	)
-	infoSchemaQueryResponseTimeTotalDesc = prometheus.NewDesc(
-		prometheus.BuildFQName(namespace, informationSchema, "query_response_time_total"),
-		"Total time of queries according to the length of time they took to execute separately.",
-		[]string{"le"}, nil,
-	)
-)
+	// Use uppercase for table names, otherwise read/write split will return the same results as total
+	// due to the bug.
+	queryResponseTimeQueries = [3]string{
+		"SELECT TIME, COUNT, TOTAL FROM INFORMATION_SCHEMA.QUERY_RESPONSE_TIME",
+		"SELECT TIME, COUNT, TOTAL FROM INFORMATION_SCHEMA.QUERY_RESPONSE_TIME_READ",
+		"SELECT TIME, COUNT, TOTAL FROM INFORMATION_SCHEMA.QUERY_RESPONSE_TIME_WRITE",
+	}
 
-// ScrapeQueryResponseTime collects from `information_schema.query_response_time`.
-func ScrapeQueryResponseTime(db *sql.DB, ch chan<- prometheus.Metric) error {
-	var queryStats uint8
-	err := db.QueryRow(queryResponseCheckQuery).Scan(&queryStats)
-	if err != nil {
-		log.Debugln("Query response time distribution is not present.")
-		return nil
+	infoSchemaQueryResponseTimeCountDescs = [3]*prometheus.Desc{
+		prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, informationSchema, "query_response_time_count"),
```

Comment:
Why the `_count` suffix? It doesn't really add any information (does it?), and it collides with the automatically added `_count` suffix in the histogram (resulting in `_count_count`).